File: cliente_cli_python/eureka_cli/api/repository_factory.py
from eureka_cli.api.environment import EnvironmentManager
from eureka_cli.api.repository import IEurekaBankRepository, RestRepository, SoapRepository


class RepositoryFactory:
    _instance: IEurekaBankRepository | None = None
    _last_server = None

    @classmethod
    def get_repository(cls) -> IEurekaBankRepository:
        current = EnvironmentManager.current_server
        if cls._instance is None or cls._last_server != current:
            if current.is_soap:
                cls._instance = SoapRepository()
            else:
                cls._instance = RestRepository()
            cls._last_server = current
        return cls._instance

File: 04.CLIMOV/app/src/main/java/com/eurekabank/api/data/repository/RepositoryFactory.kt
package com.eurekabank.api.data.repository

import com.eurekabank.api.data.network.EnvironmentManager

object RepositoryFactory {
    fun getRepository(): IEurekaBankRepository {
        return if (EnvironmentManager.currentServer.isSoap) SoapRepositoryImpl() else RestRepositoryImpl()
    }
}
